bmaptools/TransRead.py

```diff
@@ -589,6 +590,11 @@ def _print_warning(timeout):
         username = parsed_url.username
         password = parsed_url.password
 
+        auth = netrc.netrc().authenticators(parsed_url.hostname)
+        if not username and not password and auth:
+            username = auth[0]
+            password = auth[2]
```

Comment:
Could you please import netrc and create the object only if "not username and not password". It is more optimal than doing it unconditionally. Modern pylint complains about importing at a non-top level, but I'd silence that warning with a pylint directive.

Reply:
It looked like `import socket` was done a few lines before without a pylint exclusion, so I didn't think it would be necessary here; if you want I can still add it but I'm not sure how to run pylint for this project to check if it worked.